src/base.ts

```diff
@@ -41,76 +30,69 @@ const withPagination = <T>(json: {
-      break
-    case Service.CATALOGUE:
-      url = apiClient.configuration.catalogueServiceUrl
-      break
-    case Service.SEARCH:
-      url = apiClient.configuration.searchServiceUrl
-      break
-    case Service.DEALER:
-      url = apiClient.configuration.dealerServiceUrl
-      break
-    case Service.OPTION:
-      url = apiClient.configuration.optionServiceUrl
-      break
-    case Service.ANALYTICS:
-      url = apiClient.configuration.analyticsServiceUrl
-      break
-    case Service.USER_NOTIFICATION:
-      url = apiClient.configuration.userNotificationServiceUrl
-      break
-    case Service.TOKEN_REFRESH:
-      url = apiClient.configuration.tokenRefreshServiceUrl
-      break
-    default:
-      throw new Error(`Tried to resolve url of unknown service "${service}"`)
+const authorizationHeader = () => {
+  if (!apiClient.tokens.accessToken) {
+    return null
   }
 
-  if ((url || "").trim()) {
-    return url
-  }
+  return `Bearer ${apiClient.tokens.accessToken}`
+}
 
-  throw new Error(`Missing endpoint configuration for "${service}" service`)
+export interface RequestOptions {
+  headers?: Record<string, string>
+  // TODO: think about this naming
+  serviceUrl?: string
 }
 
-const authorizationHeader = () => {
-  if (!apiClient.tokens.accessToken) {
-    return null
+export interface RequestOptionsWithRecaptcha extends RequestOptions {
+  recaptchaToken?: string
+}
+
+type Method = "GET" | "POST" | "PUT" | "PATCH" | "DELETE"
+
+interface RequestOptionsWithMethod extends RequestOptions {
+  method?: Method
+}
+
+export const getHost = ({ serviceUrl }) => {
+  const url = serviceUrl || apiClient.configuration.apiGatewayUrl
+
+  if (!url) {
+    throw new Error(
+      'ApiClient not configured, please run: ApiClient.configure({ apiGatewayUrl: "your.api.gateway" }'
+    )
   }
 
-  return `Bearer ${apiClient.tokens.accessToken}`
+  return url
 }
 
-export const fetchPath = async (
-  service: Service,
-  path: string,
-  options = {}
-) => {
-  const prefix = resolveServiceUrl(service)
-  const url = `${prefix}/${stripLeadingSlash(path)}`
+export const fetchPath = async ({
+  path,
+  body,
+  options,
+}: {
+  path: string
+  body?: string
+  options: RequestOptionsWithMethod
+}) => {
+  const { headers = {}, method = "GET", serviceUrl = null } = options
+  const url = `${getHost({ serviceUrl })}/${stripLeadingSlash(path)}`
 
   if (apiClient.configuration.debug) {
     // eslint-disable-next-line no-console
     console.info(`    >> API #fetchPath: ${url}`, options)
   }
 
-  const { headers, ...otherOptions } = { headers: {}, ...options }
   const auth = authorizationHeader()
-
   const response = await fetch(url, {
     headers: {
       "Content-Type": "application/json",
       Accept: `application/vnd.carforyou.${apiClient.version}+json`,
       ...(auth ? { Authorization: auth } : {}),
       ...headers,
     },
-    ...otherOptions,
+    method,
+    ...(body ? { body } : {}),
```

Comment:
this might be intentional, but worth pointing out - this removes the possibility to pass arbitrary options to fetch. we're locking down to headers, method and body. if it was intentional, please note it in the breaking changes and verify we're not making use of that anywhere

Reply:
I'll take a look at the usage in the projects. I don't think we used that but if it's something we desire and actively use it should be easy to get back.

Reply:
Was low effort so I did it anyway